protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -483,7 +480,33 @@ func renderMessageAsDefinition(msg *descriptor.Message, reg *descriptor.Registry
 		transformAnyForJSON(&schema, reg.GetUseJSONNamesForFields())
 	}
 
-	return schema
+	return schema, nil
+}
+
+func renderFieldAsDefinition(f *descriptor.Field, reg *descriptor.Registry, refs refMap, pathParams []descriptor.Parameter) (openapiSchemaObject, error) {
+	if len(pathParams) == 0 {
+		return schemaOfField(f, reg, refs), nil
+	}
+	location := ""
+	if ix := strings.LastIndex(f.Message.FQMN(), "."); ix > 0 {
+		location = f.Message.FQMN()[0:ix]
+	}
+	msg, err := reg.LookupMsg(location, f.GetTypeName())
+	if err != nil {
+		return openapiSchemaObject{}, err
+	}
+	schema, err := renderMessageAsDefinition(msg, reg, refs, pathParams)
+	if err != nil {
+		return openapiSchemaObject{}, err
+	}
+	comments := fieldProtoComments(reg, f.Message, f)
+	if len(comments) > 0 {
+		// Use title and description from field instead of nested message if present.
+		paragraphs := strings.Split(comments, "\n\n")
```

Comment:
I'm a little confused, where does this logic come from? Are we doing this because we've already joined the title and description comments into one, and we need to separate them again? If so, is there no way to retrieve the original data so we don't have to depend on behavior like "title and descriptions are merged by double newlines"?

Reply:
This took me a while to figure out as well (my first attempt to use the description with the field comment string instead of the message comment string missed it). I generalised it from https://github.com/grpc-ecosystem/grpc-gateway/blob/09a279d441d5819789b08c0ae17544045601ca67/protoc-gen-openapiv2/internal/genopenapi/template.go#L243-L246 as it was needed in 2 places

`schema` has fields `title` and `description`, where the first paragraph of a comment string is put into the title and the rest into the description. However, `parameter` only has `description`, so we need to merge the string back.

It is split here: https://github.com/grpc-ecosystem/grpc-gateway/blob/09a279d441d5819789b08c0ae17544045601ca67/protoc-gen-openapiv2/internal/genopenapi/template.go#L1890

I changed the join from `. ` to `\n\n` for consistency, as some places the comment string is used directly, e.g. here: https://github.com/grpc-ecosystem/grpc-gateway/blob/09a279d441d5819789b08c0ae17544045601ca67/protoc-gen-openapiv2/internal/genopenapi/template.go#L986-L988

Reply:
Good idea. Done.